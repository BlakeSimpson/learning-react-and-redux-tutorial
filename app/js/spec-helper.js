import 'raf/polyfill';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

export const pointsMock = [
  {
    id: 1,
    x: 450,
    y: 110,
    details: {
      name: 'The Wall',
      house: "Night's Watch",
      words: 'Night gathers, and now my watch begins.'
    },
    favourite: true
  },
  {
    id: 2,
    x: 375,
    y: 355,
    details: {
      name: 'Winterfell',
      house: 'Stark',
      words: 'Winter is Coming'
    },
    favourite: false
  }
];
