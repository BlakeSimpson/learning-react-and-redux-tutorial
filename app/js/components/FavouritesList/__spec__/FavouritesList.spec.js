import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { pointsMock } from '../../../spec-helper';
import ConnectedFavouritesList, { FavouritesList } from '../';

const noop = () => {};

describe('FavouritesList', () => {
  describe('when there are no points', () => {
    it('matches the snapshot', () => {
      const wrapper = shallow(<FavouritesList points={[]} remove={noop} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When there are points', () => {
    it('matches the snapshot', () => {
      const wrapper = shallow(
        <FavouritesList points={pointsMock} remove={noop} />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('ConnectedApp', () => {
  const mockStore = configureStore([]);
  const store = mockStore({ points: pointsMock });

  it('maps store state to the props', () => {
    const wrapper = shallow(<ConnectedFavouritesList store={store} />);

    expect(wrapper).toMatchSnapshot();
  });
});
