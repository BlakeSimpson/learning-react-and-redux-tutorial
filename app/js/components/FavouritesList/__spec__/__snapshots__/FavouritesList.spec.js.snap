// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedApp maps store state to the props 1`] = `
<FavouritesList
  points={
    Array [
      Object {
        "details": Object {
          "house": "Night's Watch",
          "name": "The Wall",
          "words": "Night gathers, and now my watch begins.",
        },
        "favourite": true,
        "id": 1,
        "x": 450,
        "y": 110,
      },
      Object {
        "details": Object {
          "house": "Stark",
          "name": "Winterfell",
          "words": "Winter is Coming",
        },
        "favourite": false,
        "id": 2,
        "x": 375,
        "y": 355,
      },
    ]
  }
  remove={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`FavouritesList When there are points matches the snapshot 1`] = `
<div>
  <h3>
    Favourites
  </h3>
  <ul>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Ã—
      </button>
      <span>
        The Wall
      </span>
    </li>
  </ul>
</div>
`;

exports[`FavouritesList when there are no points matches the snapshot 1`] = `
<div>
  <h3>
    Favourites
  </h3>
  <ul />
</div>
`;
